test/python/test_qiskit.py

```diff
@@ -0,0 +1,197 @@
+"""Test Qiskit import."""
+
+from __future__ import annotations
+
+from typing import cast
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import AncillaRegister, ClassicalRegister, Parameter, QuantumRegister, Qubit
+from qiskit.circuit.library import MCXRecursive, MCXVChain, XXMinusYYGate, XXPlusYYGate
+from qiskit.transpiler import Layout, TranspileLayout
+
+from mqt.core._core import Expression, SymbolicOperation
+from mqt.core.qiskit_utils import qiskit_to_mqt
+
+
+def test_empty_circuit() -> None:
+    """Test import."""
+    q = QuantumCircuit()
+
+    mqt_qc = qiskit_to_mqt(q)
+    assert mqt_qc.n_qubits == 0
+    assert mqt_qc.n_ops == 0
+
+
+def test_single_gate() -> None:
+    """Test import."""
+    q = QuantumCircuit(1)
+    q.h(0)
+
+    mqt_qc = qiskit_to_mqt(q)
+    assert mqt_qc.n_qubits == 1
+    assert mqt_qc.n_ops == 1
+    assert mqt_qc[0].name.strip() == "h"
+    assert mqt_qc[0].n_qubits == 1
+
+
+def test_two_qubit_gate() -> None:
+    """Test import of two qubit gate."""
+    q = QuantumCircuit(2)
+    q.cx(0, 1)
+    mqt_qc = qiskit_to_mqt(q)
+    assert mqt_qc.n_qubits == 2
+    assert mqt_qc.n_ops == 1
+    assert mqt_qc[0].name.strip() == "x"
+    assert mqt_qc[0].n_qubits == 2
+    assert {control.qubit for control in mqt_qc[0].controls} == {0}
+
+
+def test_mcx() -> None:
+    """Test import of mcx gate."""
+    q = QuantumCircuit(3)
+    q.mcx([0, 1], 2)
+    mqt_qc = qiskit_to_mqt(q)
+    assert mqt_qc.n_qubits == 3
+    assert mqt_qc.n_ops == 1
+    assert mqt_qc[0].name.strip() == "x"
+    assert mqt_qc[0].n_qubits == 3
+    assert {control.qubit for control in mqt_qc[0].controls} == {0, 1}
+
+
+def test_mcx_recursive() -> None:
+    """Test import of large mcx gate."""
+    q = QuantumCircuit(9)
+    q.append(MCXRecursive(num_ctrl_qubits=7), range(9))
+    mqt_qc = qiskit_to_mqt(q)
+    assert mqt_qc.n_qubits == 9
+    assert mqt_qc.n_ops == 1
+    assert mqt_qc[0].name.strip() == "x"
+    assert mqt_qc[0].n_qubits == 9
```

Comment:
I ported the previous C++ implementation. No ancillary registers are allocated for this gate. Also the qiskit circuit acts on 9 non-ancillary qubits as long as I don't give it any.

Reply:
A seven-controlled MCX gate only acts on 8 qubits. IIRC the existing implementation has stripped away that one extra qubit since we can natively represent MCX gates.
Same goes for the `v-chain` mode, where we strip the `ncontrols -2` ancillary qubits to get a more compact circuit.
I can investigate this further during my review if you'd like.